Source/Csla/Configuration/ConfigurationExtensions.cs

```diff
@@ -54,13 +54,13 @@ public static IServiceCollection AddCsla(this IServiceCollection services, Actio
       services.TryAddTransient((p) => new Channels.Local.LocalProxyOptions());
       services.AddTransient<Channels.Local.LocalProxy, Channels.Local.LocalProxy>();
       // Add server-side data portal; this is always necessary to support RunLocal
-      cslaOptions.DataPortal().ApplyConfiguration();
+      cslaOptions.DataPortal(options => options.ApplyConfiguration());
 
       // Default to using LocalProxy and local data portal
       var proxyInit = services.Where(i => i.ServiceType.Equals(typeof(IDataPortalProxy))).Any();
       if (!proxyInit)
       {
-        cslaOptions.DataPortal().UseLocalProxy();
+        cslaOptions.DataPortal((options) => options.UseLocalProxy());
       }
```

Comment:
As above, I think there is a difference between how it is accessed by the end user and how we access it internally. There is nothing wrong with this code, but there is something about the way it reads that makes it look like we are mixing up config from application of config.

Most of the time, mixing the two concepts up doesn't matter; TryAdd<blah> sorts out the confusion. However, occasionally, it does matter if you mix them up. The interceptors code didn't work when we mixed these two concepts. In that scenario we are registering multiple types, and calling the same code twice caused each type to be registered twice; instead of 2 interceptors, we had 4.

Reply:
Not anymore - now end users will be writing this code too. Because the `DataPortal` method no longer returns a unique type, we can't write extension methods over it. It now returns a `CslaOptions` instance and so resets to the base level of extension methods.

Reply:
Sorry, I think my comment was maybe a bit ambiguous. I think what I meant was that the following might read better from _within_ the framework:

```csharp
cslaOptions.DataPortalClientOptions.UseLocalProxy();
```
 
I know this has changed anyway, but the point was about the usage/selection of the coding styles. The lambdas look like they set config and are available to end users, and the direct method call looks like it applies that config and can only be done inside of the framework.

No matter.